sda/internal/database/db_functions.go

```diff
@@ -62,6 +62,42 @@ func (dbs *SDAdb) getFileID(corrID string) (string, error) {
 	return fileID, nil
 }
 
+// GetInboxFilePathFromID checks if a file exists in the database for a given user and fileID
+// and that is not yet archived
+func (dbs *SDAdb) GetInboxFilePathFromID(submissionUser, fileID string) (string, error) {
+	var (
+		err      error
+		count    int
+		filePath string
+	)
+
+	for count == 0 || (err != nil && count < RetryTimes) {
+		filePath, err = dbs.getInboxFilePathFromID(submissionUser, fileID)
+		count++
+	}
+
+	return filePath, err
+}
+
+func (dbs *SDAdb) getInboxFilePathFromID(submissionUser, fileID string) (string, error) {
+	dbs.checkAndReconnectIfNeeded()
+	db := dbs.DB
+
+	const getFilePath = "SELECT submission_file_path from sda.files where " +
+		"submission_user= $1 and id = $2 " +
+		"AND EXISTS (SELECT 1 FROM " +
+		"(SELECT event from sda.file_event_log where file_id = $2 order by started_at desc limit 1) " +
+		"as subquery WHERE event in ('registered', 'uploaded', 'submitted', 'ingested', 'error'))"
```

Comment:
This will be a bit iffy since in FEGA if the submission user adds a file to a run by mistake that file will be ingest, archived, backed up and given a stable ID (a case we actually will ned to be able to handle).
If we want to limit this to files that only have been uploaded to the inbox `ingested` can not be a valid event since then the file will also be in the archive.

Instead we probably need to check if the file has had a state of `ingested`, `verified`,  `archived` or `ready` and then also remove it from the archive. If the file has had the state `backed up` we also need to remove the file from the backup site.

Reply:
If we focus on files that are strictly uploaded only 'uploaded' would be the available option.

Reply:
Fixed in the latest commit.